pkg/asset/machines/aws/worker.go

```diff
@@ -2,98 +2,124 @@
 package aws
 
 import (
-	"text/template"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1alpha1"
+	clusterapi "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 
 	"github.com/openshift/installer/pkg/types"
 )
 
-// WorkerConfig is used to generate the worker machinesets.
-type WorkerConfig struct {
-	Instances []WorkerMachinesetInstance
-	MachineConfig
-}
+// NonMasterMachineSets returns a list of machinesets for a machinepool.
+func NonMasterMachineSets(config *types.InstallConfig, pool *types.MachinePool, userdatasecret string) ([]clusterapi.MachineSet, error) {
+	var out []clusterapi.MachineSet
+	if pool.Name == "master" {
+		return nil, fmt.Errorf("pool name master is not supported")
+	}
+	if config.Platform.AWS == nil || pool.Platform.AWS == nil {
+		return nil, fmt.Errorf("invalid inputs")
+	}
+	clustername := config.ObjectMeta.Name
+	platform := config.Platform.AWS
+	mpool := pool.Platform.AWS
+	azs := mpool.Zones
```

Comment:
So this assumes the config's AWS platform config always specifies a list of AZs. So in Hive, we will not require that, but we'll have to lookup all AZs when it is not set, and transfer that to the platform config before handing off to this function. Does this seem preferable to keeping the two concepts separate, what the user asked for, and a list of all AZs passed directly to the function and let the function figure out what to do with it?

Can work it either way but it's worth asking if we should preserve the concept of what the config asked for more clearly.